From 1f2968430df446dcbf0e27ff870ac7aa6e8113cd Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Tue, 6 Mar 2012 16:11:01 +0100
Subject: [PATCH] Fix library check when updating GCC.

As it only happens on ARM, the new GCC libdir is not found.
---
 tools/quality-agent.d/003-libs-location |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/tools/quality-agent.d/003-libs-location b/tools/quality-agent.d/003-libs-location
index 48b2521..ce5ba72 100755
--- a/tools/quality-agent.d/003-libs-location
+++ b/tools/quality-agent.d/003-libs-location
@@ -16,13 +16,16 @@ esac
 
 # Find gcc libdir.
 gcc_libdir=$(gcc -print-libgcc-file-name)
-gcc_libdir=$(dirname ${gcc_libdir})
+for i in 1 2; do
+	gcc_libdir=$(dirname ${gcc_libdir})
+done
+gcc_libdir=$(ls ${BUILDROOT}${gcc_libdir}/* 2>/dev/null | tail -n1)
 
 function check() {
 	local failed=0
 	local found
 
-	for lib in $(find ${BUILDROOT}/${libdir} -type f -name "lib*.so.*" 2>/dev/null); do
+	for lib in $(find ${BUILDROOT}/{,usr/}${libdir} -type f -name "lib*.so.*" 2>/dev/null); do
 		lib=${lib##*/}
 		lib=${lib%%.so*}
 
-- 
1.7.3.4

